code-reviews/recipes/azure-pipelines-yaml.md

```diff
@@ -0,0 +1,119 @@
+
+## Code Analysis / Linting
+
+The most popular YAML linter is [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) extension. This extension provides YAML validation, document outlining, Auto completion, Hover support and Formatter features.
+
+## VS Code Extensions
+
+There is an [Azure Pipelines for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-azure-devops.azure-pipelines) extension to add syntax highlighting and autocompletion for Azure Pipelines YAML to VS Code. It also helps you set up continuous build and deployment for Azure WebApps without leaving VS Code.
+
+## YAML in Azure Pipelines Overview
+
+When the pipeline is triggered, before running the pipeline, there are some different phases such as `Queue Time`, `Compile Time` and `Runtime`, in these phases variables are interpretted by their [runtime expression syntax](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#runtime-expression-syntax).
+
+When pipeline is triggered all nested YAML files are expanded to run in Azure Pipelines. This check list visits some tips and tricks for reviewing all nested YAML files.
+
+Before starting review please see the following documents:
+
+- Make sure to check [Azure Pipelines YAML documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema).
+
+- Understand how pipelines work [Pipeline run sequence](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runs?view=azure-devops)
+
+- Get familiar with [Key concepts for new Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/key-pipelines-concepts?view=azure-devops)
+
+  **Key concepts overview**
+![Azure Pipelines key concepts](images/key-concepts-overview.png)
+
+  - A trigger tells a Pipeline to run.
+  - A pipeline is made up of one or more stages. A pipeline can deploy to one or more environments.
+  - A stage is a way of organizing jobs in a pipeline and each stage can have one or more jobs.
+  - Each job runs on one agent. A job can also be agentless.
+  - Each agent runs a job that contains one or more steps.
+  - A step can be a task or script and is the smallest building block of a pipeline.
+  - A task is a pre-packaged script that performs an action, such as invoking a REST API or publishing a build artifact.
+  - An artifact is a collection of files or packages published by a run.
+
+## Code Review Checklist
+
+In addition to the [Code Review Checklist](../process-guidance/reviewer-guidance.md) you should also look for these Azure Pipelines YAML specific code review items.
+
+### Pipeline Structure
+
+- [ ] First understand [Pipeline structure](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pipeline-structure) and identify components
```

Comment:
This looks more like a set of steps than a checklist to verify - can we rename the items to something that can be checked, i.e. 

- [ ] The steps are well understood and components are easily identifiable

or make it into an ordered or unordered list (if they don't need to be in order)

Reply:
Thanks @TessFerrandez. Sure, I changed to your suggestion. It's more a checklist, it's better to keep as it is.